### ISSUE TITLE
Indent Fix Doesn't Shift Inline Comments with Arrow Function Body

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#8445]
**Tell us about your environment**

* **ESLint Version:** 4.0.0-alpha.0
* **Node Version:** 7.7.4
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
{
    "extends": "eslint:recommended",
    "rules": {
        "indent": 2,
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
a()
.catch((err) => {
    // docs
});
```

**What did you expect to happen?**
```js
a()
    .catch((err) => {
        // docs
    });
```
**What actually happened? Please include the actual, raw output from ESLint.**
```js
a()
    .catch((err) => {
    // docs
    });
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I think this is working as intended -- the rule allows comments to be indented at the same level as the line below or above it, to allow for comments like this:

```js
a()
    .catch((err) => {

    // comment about the foo() call
    }).then((result) => foo(result));
```

The rule can't tell what code the comment is supposed to be referring to, so it allows several possible indentations for the comment.

**Comment by epmatsw** [comment]
Ah, interesting. That does make sense.

**Comment by not-an-aardvark** [comment]
Closing this issue because the behavior is working as intended.
